### PR TITLE
Fix data simulation functions sample size specification

### PR DESCRIPTION
**Pull request by ntorresd**
This PR corrects the data simulation functions `get_sim_probability`, `get_sim_n_seropositive` and `generate_sim_data` for them to build the simulated data based on the desired age structure of the survey and the desired sample sizes for each age group. 

The use of `generate_sim_data`  is exemplified by:
```
n_years <- 50
sim_data <- data.frame(
  age_mean_f = seq(1,n_years),
  tsur = 2050
)
foi <- rep(0.02, n_years)
sample_size_by_age <- as.integer(runif(n = n_years, 5, 10))
sim_data <- generate_sim_data(
  sim_data = sim_data,
  foi = foi,
  sample_size_by_age = sample_size_by_age,
  survey_label = "sim_constant_foi"
)
```
which generates a serosurvey with random sample sizes for each specified age group [1, ..., 50] (see image below). 

image

### CODE REVIEW

File: R/seroprevalence_data.R [hunk 1]
```diff
@@ -243,52 +239,46 @@ get_sim_n_seropositive <- function(
...
-    sim_data <- sim_data %>%
-        mutate(counts = sim_n_seropositive$n_seropositive,
-               total = sample_size_by_age) %>%
-        prepare_serodata()
+) {
+  sim_data <- sim_data %>%
+    mutate(
+      country = "None",
+      test = "test",
+      antibody = "antibody",
```
Comment (ben18785) [inline R/seroprevalence_data.R:270]: Do we need to set these?

File: R/modelling.R [hunk 2]
```diff
@@ -435,7 +435,13 @@ get_prev_expanded <- function(foi,
 
   prev_pn <- t(1 - exp(-exposure_expanded %*% t(foi_expanded)))
 
-  predicted_prev <- t(apply(prev_pn, 2, function(x) quantile(x, c(0.5, 0.1, 0.9))))
+  predicted_prev <- t(
+    apply(
+      prev_pn,
+      2,
+      function(x) quantile(x, c(0.5, 0.1, 0.9))
```
[the remark this answers is not in the document]
Comment (ntorresd) [reply R/modelling.R:442]: Yes! I think this is a good idea. I'll add them as optional parameters both here and in the corresponding plotting functions.

File: R/seroprevalence_data.R (hunk 1 above)
Comment (ntorresd) [reply R/seroprevalence_data.R:270]: Not really. These variables are optional and are only used in some specific instances (e.g. `extract_seromodel_summary`) just to show additional information about the serosurveys. Removing them will trigger a warning in the tests once the simulated data is passed to `prepare_serodata()` and assign the value `"None"` to them (see #144 ), so I think is okay to remove them to simplify the simulation functions as much as possible.

Comment (ntorresd) [reply R/seroprevalence_data.R]: Addressed by [92f47b8](https://github.com/epiverse-trace/serofoi/pull/140/commits/92f47b82bd95e42b3abd03cbbfff7e111f2383d6)

File: R/modelling.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (ntorresd) [reply R/modelling.R]: Addressed by [1aa4508](https://github.com/epiverse-trace/serofoi/pull/140/commits/1aa4508858961d90506a5441b8488a301dc0e18d) and subsequent commits (documentation only).

**Review by ben18785 — APPROVED** [review]
Looks great -- thanks @ntorresd

**Pull request merged**